code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -173,3 +173,25 @@
 		else
 			if(decal.random_icon_states && length(decal.random_icon_states) > 0) //If there's already one, just randomize it so it changes.
 				decal.icon_state = pick(decal.random_icon_states)
+
+		var/splash_chance = 40 //Base chance of getting splashed. Decreases with # of victims.
+		var/distance = 0 //Distance, decreases splash chance.
+		var/i = 0 //Tally up our victims.
+		
+		if(!(xeno_caste.caste_flags & CASTE_ACID_BLOOD))
+			return
+		else
```

Comment:
No need for an `else` if you return, but on this kind of proc it's better to have this within the `if`, as more things can happen after.
Also the definition of variables such as `splash_chance`, `distance` and `i` should be within the `if`, as they are not used anywhere else.